### PR TITLE
lint: add license headers to `staticcheck` files in `build` dir

### PR DESCRIPTION
**Pull request by rickystewart**
... and update linters to evaluate what's inside the `build` dir as well.

Part of: RE-658
Epic: DEVINF-1246
Release note: None
Release justification: Non-production code changes

### CODE REVIEW

**Review by jlinder — CHANGES_REQUESTED** [review]
One change please.

**Comment by jlinder** [comment]
Also a note:

- The `Epic: DEVINF-1246` line isn't needed in the PR or commit when there is a `Part of:` issue reference.

File: pkg/cmd/generate-staticcheck/main.go [hunk 1]
```diff
@@ -22,8 +22,13 @@ const (
 `
 	rootBuildBazelContent = `exports_files(["def.bzl"])
 `
-	analysisFileTemplate = `// Code generated by generate-staticcheck; DO NOT EDIT.
-
+	analysisFileTemplate = `// Copyright 2016 The Cockroach Authors.
```
[the remark this answers is not in the document]
Comment (rickystewart) [reply pkg/cmd/generate-staticcheck/main.go:25]: Oops, bad copy-paste.

**Review by jlinder — APPROVED** [review]
There's a go formatting lint error. GTG once that's fixed.

**Comment by rickystewart** [comment]
TFTR! 

bors r=jlinder

**Pull request merged**